components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserSelfRegistrationHandler.java

```diff
@@ -438,12 +448,20 @@ private void triggerNotification(User user, String notificationChannel, String c
             for (Property prop : props) {
                 properties.put(prop.getKey(), prop.getValue());
             }
+        }if (StringUtils.isNotBlank(code)) {
+            if (emailOTPenabled) {
+                properties.put(IdentityRecoveryConstants.OTP_CODE, code);
+            } else {
+                properties.put(IdentityRecoveryConstants.CONFIRMATION_CODE, code);
+            }
         }
-        if (StringUtils.isNotBlank(code)) {
-            properties.put(IdentityRecoveryConstants.CONFIRMATION_CODE, code);
-        }
-        properties.put(IdentityRecoveryConstants.TEMPLATE_TYPE,
+        if (emailOTPenabled) {
```

Comment:
Check for the canhandle method to which events it can handle

Reply:
The method is used only for the self user registration purpose. So we do not have to remove this logic out of TriggerNotification method.